src/OpenRiaServices.DomainServices.Client/Test/Client.Test/Data/ServerSideAsyncTests.cs

```diff
@@ -330,5 +330,167 @@ public void Invoke_TaskAsyncWithException_InTask()
+
+            // Act, Verify
+            await ctx.SubmitChangesAsync();
+            Assert.AreEqual("updated", rangeItem.Text);
+        }
+
+        [TestMethod]
+        [Description("Test that exceptions thrown in Task from Task async Update methods are propagated")]
+        public async Task Update_TaskAsyncWithException_InTask()
+        {
+            // Setup
+            var ctx = new ServerSideAsyncDomainContext(TestURIs.ServerSideAsync);
+            var rangeItem = new RangeItem() { ThrowDomainException = true };
+            ctx.RangeItems.Attach(rangeItem);
+
+            rangeItem.Text = "notupdating";
+
+            // Act
+            var domainException = await ExceptionHelper.ExpectExceptionAsync<DomainException>(ctx.SubmitChangesAsync, "UpdateRangeAsync");
+
+            // Verify
+            Assert.AreEqual(26, domainException.ErrorCode, "Wrong error code returned or expected operation was not executed");
+            Assert.AreEqual("notupdating", rangeItem.Text);
+        }
+
+        [TestMethod]
+        [Description("Tests that an async Task EntityAction operation is executed")]
+        public async Task EntityAction_TaskAsync()
+        {
+            // Setup
+            var ctx = new ServerSideAsyncDomainContext(TestURIs.ServerSideAsync);
+            var rangeItem = new RangeItem();
+            ctx.RangeItems.Attach(rangeItem);
+
+            rangeItem.Text = "updating";
+            rangeItem.CustomUpdateRange();
+
+            // Act, Verify
+            await ctx.SubmitChangesAsync();
+            Assert.AreEqual("custom updated", rangeItem.Text);
+        }
+
+        [TestMethod]
+        [Description("Test that exceptions thrown in Task from Task async EntityAction methods are propagated")]
+        public async Task EntityAction_TaskAsyncWithException_InTask()
+        {
+            // Setup
+            var ctx = new ServerSideAsyncDomainContext(TestURIs.ServerSideAsync);
+            var rangeItem = new RangeItem() { ThrowDomainException = true };
+            ctx.RangeItems.Attach(rangeItem);
+
+            rangeItem.CustomUpdateRange();
+
+            // Act
+            var domainException = await ExceptionHelper.ExpectExceptionAsync<DomainException>(ctx.SubmitChangesAsync, "CustomUpdateRangeAsync");
+
+            // Verify
+            Assert.AreEqual(28, domainException.ErrorCode, "Wrong error code returned or expected operation was not executed");
+        }
+
+        [TestMethod]
+        [Description("Tests that an async Task Delete operation is executed and waited for")]
+        public async Task Delete_TaskAsync()
+        {
+            // Setup
+            var ctx = new ServerSideAsyncDomainContext(TestURIs.ServerSideAsync);
+            var rangeItem = new RangeItem() { Id = 42 };
+            ctx.RangeItems.Attach(rangeItem);
+
+            ctx.RangeItems.Remove(rangeItem);
+
+            // Act, Verify
+            await ctx.SubmitChangesAsync();
+        }
```

Comment:
I had another look at this, we could validate that `ChangeSetResult.RemovedEntities` contains rangeItem 

Eg. the collection should be equal to `new [] {rangeItem}`